tests/e2e/monitoring_test.go

```diff
@@ -1093,3 +1106,480 @@ func withMonitoringTraces(backend, secret, size, retention string) testf.Transfo
 
 	return testf.TransformPipeline(transforms...)
 }
+
+// ValidateNamespaceRestrictedMetricsAccess tests the namespace-restricted metrics access functionality.
+func (tc *MonitoringTestCtx) ValidateNamespaceRestrictedMetricsAccess(t *testing.T) {
+	t.Helper()
+
+	dsci := tc.FetchDSCInitialization()
+
+	// Ensure metrics are configured
+	tc.EventuallyResourceCreatedOrUpdated(
+		WithMinimalObject(gvk.DSCInitialization, tc.DSCInitializationNamespacedName),
+		WithMutateFunc(testf.TransformPipeline(
+			testf.Transform(`.spec.monitoring.managementState = "%s"`, operatorv1.Managed),
+			withMetricsConfig(),
+		)),
+	)
+
+	// Verify the prometheus-restricted deployment is created
+	tc.EnsureResourceExists(
+		WithMinimalObject(gvk.Deployment, types.NamespacedName{
+			Name:      "prometheus-restricted",
+			Namespace: dsci.Spec.Monitoring.Namespace,
+		}),
+		WithCondition(And(
+			jq.Match(`.status.readyReplicas == 1`),
+			jq.Match(`.spec.template.spec.containers | length == 2`), // kube-rbac-proxy + prom-label-proxy
+		)),
+		WithCustomErrorMsg("prometheus-restricted deployment should be created and ready"),
+	)
+
+	// Verify the service account exists with proper RBAC
+	tc.EnsureResourceExists(
+		WithMinimalObject(gvk.ServiceAccount, types.NamespacedName{
+			Name:      "prometheus-restricted",
+			Namespace: dsci.Spec.Monitoring.Namespace,
+		}),
+	)
+
+	// Verify the ClusterRoleBinding exists for prometheus metrics reader
+	tc.EnsureResourceExists(
+		WithMinimalObject(gvk.ClusterRoleBinding, types.NamespacedName{
+			Name: "prometheus-restricted",
+		}),
+		WithCondition(And(
+			jq.Match(`.roleRef.name == "odh:prometheus-metrics-reader"`),
+			jq.Match(`.subjects[0].name == "prometheus-restricted"`),
+			jq.Match(`.subjects[0].namespace == "%s"`, dsci.Spec.Monitoring.Namespace),
+		)),
+	)
+
+	// Verify the service is created with proper annotations for TLS
+	tc.EnsureResourceExists(
+		WithMinimalObject(gvk.Service, types.NamespacedName{
+			Name:      "prometheus-restricted",
+			Namespace: dsci.Spec.Monitoring.Namespace,
+		}),
+		WithCondition(And(
+			jq.Match(`.spec.ports[0].port == 8443`),
+			jq.Match(`.spec.ports[0].name == "https"`),
+			jq.Match(`.metadata.annotations."service.beta.openshift.io/serving-cert-secret-name" == "prometheus-restricted-tls"`),
+		)),
+	)
+
+	// Verify the route is created for external access
+	tc.EnsureResourceExists(
+		WithMinimalObject(gvk.Route, types.NamespacedName{
+			Name:      "data-science-prometheus-route",
+			Namespace: dsci.Spec.Monitoring.Namespace,
+		}),
+		WithCondition(And(
+			jq.Match(`.spec.to.name == "prometheus-restricted"`),
+			jq.Match(`.spec.tls.termination == "reencrypt"`),
+			jq.Match(`.spec.tls.insecureEdgeTerminationPolicy == "Redirect"`),
+		)),
+	)
+}
+
+// ValidatePrometheusSecureProxyAuthentication tests the Prometheus secure proxy authentication and authorization.
+func (tc *MonitoringTestCtx) ValidatePrometheusSecureProxyAuthentication(t *testing.T) {
```

Comment:
Some of the checks here are duplicated from the previous method. I think they can be merged.